app/controllers/registrations_controller.rb

```diff
@@ -1,6 +1,38 @@
 class RegistrationsController < Devise::RegistrationsController
+  def update
+    update_params = account_update_params
+    # required for settings form to submit when password is left blank
+    if update_params[:password].blank?
+      [:password,:password_confirmation,:current_password].collect{|p| update_params.delete(p) }
+    end
+    if update_params[:password].present? && update_params[:current_password].blank?
+      [:current_password].collect{|p| update_params.delete(p) }
```

Comment:
Prefer `map` over `collect`.<br>Space missing to the left of {.<br>Space between { and | missing.